tds/src/main/java/thredds/server/viewer/ViewerServiceImpl.java

```diff
@@ -210,6 +225,30 @@ private Access getOpendapAccess(Dataset ds) {
     }
   }
 
+  private static class JupyterNotebookViewer implements Viewer {
+    private static final String title = "Jupyter Notebook viewer";
+
+    public boolean isViewable(Dataset ds) {
+      return true;
+    }
```

Comment:
Right now, TDS treats all datasets as viewable with Jupyter Notebooks as long as there is at least one ipynb file in the content directory, but we don't check for NB compatibility with the dataset until the NotebookController tries to return a file, which means some datasets might link to Jupyter Notebook viewer but not be able to return a valid notebook :/ Not sure whether to move that logic to here (but then it would execute on page load rather than download which seems wonky), or do something like return an empty or mostly empty file.